From 069610b635462f69767f7e9d2891e848ed749672 Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@freescale.com>
Date: Mon, 17 Jun 2013 15:10:03 +0300
Subject: [PATCH 0395/1089] dpaa_eth: fix issue in dpa_fq_setup()

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Only use the dpa_fq_list entries of type FQ_TYPE_TX as egress FQs.

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
Change-Id: I3310c2461e09b539a223d4c676b8b8ebe1b133d3
Reviewed-on: http://git.am.freescale.net:8181/3088
Reviewed-by: Sovaiala Cristian-Constantin-B39531 <Cristian.Sovaiala@freescale.com>
Reviewed-by: Radulescu Ruxandra Ioana-B05472 <ruxandra.radulescu@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
index a6e0df1d..1f731d6 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
@@ -2911,6 +2911,8 @@ void dpa_fq_setup(struct dpa_priv_s *priv, const dpa_fq_cbs_t *fq_cbs,
 	 */
 	while (egress_cnt < DPAA_ETH_TX_QUEUES) {
 		list_for_each_entry(fq, &priv->dpa_fq_list, list) {
+			if (fq->fq_type != FQ_TYPE_TX)
+				continue;
 			priv->egress_fqs[egress_cnt++] = &fq->fq_base;
 			if (egress_cnt == DPAA_ETH_TX_QUEUES)
 				break;
-- 
2.0.2

